### PR TITLE
add json interpreter

### LINKED ISSUE #21: json interpreter

### LINKED ISSUE #44: populate current calendar, days, events onto view
load the current month onto the calendar view.... this is assuming that the json file is read already and the calendar object populated with events/tasks already exist

For example, when the user first opens up their calendar, it should open up to a default month. What you should do is to change the calendar to reflect the month, update the days, put the correct order and make sure days match with days of the week. Then in the calendar object, check if any events/tasks exist in this month. Then add them to the calendar.

NOTE THERE ARE FUNCTIONS THAT RETURN IF A MONTH IS A LEAP YEAR GIVEN YEAR, HOW MANY DAYS ARE IN A MONTH GIVEN YEAR AND MONTH, AND WHAT DAY OF THE WEEK IT IS GIVEN YEAR MONTH AND DAY. 

**LOCATED IN 'calendar_helper.js'**

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by stevenkhaw** [comment]
No not merge yet, i have a few questions and just want to confirm a few things

File: source/assets/scripts/Year.js [hunk 1]
```diff
@@ -19,9 +19,9 @@ class Year {
      * @param {boolean} isLeapYear boolean on if leap year
      * @param {Month[]} months months of year 
      */
-    constructor(currYear, isLeapYear, months) {
+    constructor(currYear, months) {
         this.currYear = currYear;
-        this.isLeapYear = isLeapYear;
+        this.isLeapYear = ((0 === this.curYear % 4) && (0 !== this.curYear % 100) || (0 === this.curYear % 400));
```
Comment (stevenkhaw) [inline source/assets/scripts/Year.js:24]: curYear is not a variable, currYear is a variable

File: source/assets/scripts/jsonInterpreter.js [hunk 2]
```diff
@@ -1,143 +1,145 @@
...
-  console.log(events_list.length);
-  
-  refreshCalendar();
-
-  initFormHandler();
+`;
+
+/**
+ * Get the corresponding string of a day 
+ * @param ind - order number of the day (1 - 7)
```
Comment (stevenkhaw) [inline source/assets/scripts/jsonInterpreter.js]: JSDoc comments should also have a {number} so we can type check the variable. In this case line 62 should be @param {number} ind - order number of the day (1 - 7)

File: source/assets/scripts/Year.js (hunk 1 above)
Comment (stevenkhaw) [inline source/assets/scripts/Year.js]: curYear is not a variable, currYear is a variable

File: source/assets/scripts/jsonInterpreter.js (hunk 2 above)
Comment (stevenkhaw) [reply source/assets/scripts/jsonInterpreter.js]: do this for all of @param

File: source/assets/scripts/jsonInterpreter.js [hunk 3]
```diff
@@ -1,143 +1,145 @@
...
+ * @param description - the description of event
```
Comment (stevenkhaw) [inline source/assets/scripts/jsonInterpreter.js]: what is param description?

**Review by stevenkhaw — CHANGES_REQUESTED** [review]
check comments

**Comment by ghost** [comment]
Added @param type for getMonthStr and getDayStr; fixed variable name to currYear; removed misplaced @param description

**Review by stevenkhaw — APPROVED** [review]

**Pull request merged**